File: test_projects/c_project/calculator.c
#include <stdio.h>
#include "calculator.h"

void init_calculator(Calculator* calc) {
    calc->count = 0;
}

double add(Calculator* calc, double a, double b) {
    double result = a + b;
    if (calc->count < MAX_HISTORY) {
        calc->history[calc->count].a = a;
        calc->history[calc->count].b = b;
        calc->history[calc->count].result = result;
        calc->count++;
    }
    return result;
}

double subtract(Calculator* calc, double a, double b) {
    double result = a - b;
    if (calc->count < MAX_HISTORY) {
        calc->history[calc->count].a = a;
        calc->history[calc->count].b = b;
        calc->history[calc->count].result = result;
        calc->count++;
    }
    return result;
}

double multiply(Calculator* calc, double a, double b) {
    double result = a * b;
    if (calc->count < MAX_HISTORY) {
        calc->history[calc->count].a = a;
        calc->history[calc->count].b = b;
        calc->history[calc->count].result = result;
        calc->count++;
    }
    return result;
}

double divide(Calculator* calc, double a, double b) {
    if (b == 0.0) {
        printf("Error: Division by zero\n");
        return 0.0;
    }
    double result = a / b;
    if (calc->count < MAX_HISTORY) {
        calc->history[calc->count].a = a;
        calc->history[calc->count].b = b;
        calc->history[calc->count].result = result;
        calc->count++;
    }
    return result;
}

void print_history(const Calculator* calc) {
    printf("\nCalculation History:\n");
    for (int i = 0; i < calc->count; i++) {
        printf("%.2f + %.2f = %.2f\n", 
               calc->history[i].a, 
               calc->history[i].b, 
               calc->history[i].result);
    }
}

File: test_projects/python_project/calculator.py
"""计算器模块 - 实现基本数学运算"""


class Calculator:
    """简单的计算器类"""

    def __init__(self):
        """初始化计算器"""
        self.history = []

    def add(self, a, b):
        """加法运算"""
        result = a + b
        self.history.append(f"{a} + {b} = {result}")
        return result

    def subtract(self, a, b):
        """减法运算"""
        result = a - b
        self.history.append(f"{a} - {b} = {result}")
        return result

    def multiply(self, a, b):
        """乘法运算"""
        result = a * b
        self.history.append(f"{a} * {b} = {result}")
        return result

    def divide(self, a, b):
        """除法运算"""
        if b == 0:
            raise ValueError("Division by zero")
        result = a / b
        self.history.append(f"{a} / {b} = {result}")
        return result

    def get_history(self):
        """获取计算历史"""
        return self.history.copy()
